AmplifyPlugins/Storage/AWSS3StoragePluginTests/Operation/AWSS3StorageUploadFileOperationTests.swift

```diff
@@ -13,6 +13,47 @@ import XCTest
 import AWSS3
 
 class AWSS3StorageUploadFileOperationTests: AWSS3StorageOperationTestBase {
+    
+    func testUploadFileOperationAccessDenied() throws {
+        let path = NSTemporaryDirectory().appending(UUID().uuidString)
+        let key = (path as NSString).lastPathComponent
+        let url = URL(fileURLWithPath: path)
+        let contents = Data(UUID().uuidString.utf8)
+        FileManager.default.createFile(atPath: path, contents: contents)
+        try FileManager.default.setAttributes([FileAttributeKey.posixPermissions: 000], ofItemAtPath: path)
+
+        let options = StorageUploadFileRequest.Options(accessLevel: .protected)
+        let request = StorageUploadFileRequest(key: key, local: url, options: options)
+
+        let progressExpectation = expectation(description: "progress")
+        progressExpectation.isInverted = true
+        let progressListner: ProgressListener = { progress in
+            progressExpectation.fulfill()
+        }
+        let resultExpectation = expectation(description: "result")
+        let resultListener: StorageUploadFileOperation.ResultListener = { result in
+            switch result {
+            case .failure(let error):
+                guard case .accessDenied = error else {
+                    XCTFail("Should have failed with validation error")
+                    return
+                }
+                resultExpectation.fulfill()
+            case .success:
+                XCTFail("Expecting an error but got success")
+            }
+        }
+        let operation = AWSS3StorageUploadFileOperation(request,
+                                                        storageConfiguration: testStorageConfiguration,
+                                                        storageService: mockStorageService,
+                                                        authService: mockAuthService,
+                                                        progressListener: progressListner,
+                                                        resultListener)
+        operation.start()
+
+        wait(for: [progressExpectation, resultExpectation], timeout: 1)
```

Comment:
can we add a constant for timeout or use an existing one?

Reply:
Normally, I would say yes, but:

1. We already do it this way throughout all unit tests in this target.
2. Since this is a well-known API and the `timeout:` part of the selector helps any new folks, I think adding a constant hinders readability (because a reader would have to scan back to where that is defined, specially if not named something like `oneSecond`).

What do you think? You feel strongly about this?